lib/functions/ui.functions.sh

```diff
@@ -37,6 +45,10 @@ toggle_spin() {
                 done
             ) &
             spin_pid=$!
+            process_trap_add INT '_ui_trap_sig'
+            if [ ! $(process_trap_get_exit INT) ]; then
```

Comment:
This can be simplified to:

```
if ! process_trap_get_exit INT >/dev/null; then
```

This also avoids creating an additional subshell.